addons/mail/models/mail_activity_mixin.py

```diff
@@ -235,12 +235,28 @@ def _search_my_activity_date_deadline(self, operator, operand):
         return [('activity_ids', 'in', activity_ids)]
 
     def write(self, vals):
+        """Trigger res_name update on related activities, as we cannot depend on arbitrary res_model and res_id in compute methods.
+
+        As a result display_name computation for models implementing this mixin should be kept simple.
+        """
         # Delete activities of archived record.
         if 'active' in vals and vals['active'] is False:
             self.env['mail.activity'].sudo().search(
                 [('res_model', '=', self._name), ('res_id', 'in', self.ids)]
             ).unlink()
-        return super(MailActivityMixin, self).write(vals)
+        original_display_names = self.mapped('display_name')
+        write_res = super().write(vals)
+        # flush necessary to get the right compute order for 'account.move'
+        # and avoid mistakenly having it behave as though we set the name manually...
+        # see AccountMove.write where flush_recordset is also performed
+        self.flush_recordset()
```

Comment:
This sucks but I've reached the limits of how long I'm willing to trace low-level compute logic.
Compute methods shouldn't be doing all the stuff they do in account moves, but I don't have another solution other than entirely excluding the model from this logic (which could be ok..)

Impacted test is `test_sequence_concurency`, which starts a transaction before another one is comitted, and ensures that the original transaction can still complete without hitting the "unique name constraint" (using account's custom fancy sequence mixin)

Reply:
Either you flush for every model, or we could also only flush when we have an `account.move`. This model is extremely complex and their write is a nightmare. (Almost 4000 lines for a singular model, feckin hell.)

Kinda your call on this one, I'd also be willing to simply willing to exclude your logic for `account.move` but the flush when it's the problematic model could also be a solution.